docs/ai/how-to/snippets/semantic-kernel/LocalModelExamples.cs

```diff
@@ -0,0 +1,127 @@
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.SemanticKernel;
+using Microsoft.SemanticKernel.ChatCompletion;
+using Microsoft.SemanticKernel.TextGeneration;
+
+class LocalModelExamples
+{
+    public static async Task Examples()
+    {
+        AddTextGenerationServiceExample();
+        AddChatCompletionServiceExample();
+        await UseTextGenerationServiceExample();
+        await UseChatCompletionServiceExample();
+    }
+
+    static Kernel AddTextGenerationServiceExample()
+    {
+        // <addTextService>
+        IKernelBuilder builder = Kernel.CreateBuilder();
+
+        // Add your text generation service as a singleton instance
+        var service = new MyTextGenerationService();
+        builder.Services.AddKeyedSingleton<ITextGenerationService>("myTextService1", service);
+
+        // Alternatively, add your text generation service as a factory method
+        builder.Services.AddKeyedSingleton<ITextGenerationService>("myTextService2", (_, _) => new MyTextGenerationService());
+
+        // Add any other Kernel services or configurations
+        // ...
+        Kernel kernel = builder.Build();
+        // </addTextService>
+
+        return builder.Build();
+    }
+
+    static Kernel AddChatCompletionServiceExample()
+    {
+        // <addChatService>
+        IKernelBuilder builder = Kernel.CreateBuilder();
+
+        // Add your chat completion service as a singleton instance
+        var service = new MyChatCompletionService();
+        builder.Services.AddKeyedSingleton<IChatCompletionService>("myChatService1", service);
+
+        // Alternatively, add your chat completion service as a factory method
+        builder.Services.AddKeyedSingleton<IChatCompletionService>("myChatService2", (_, _) => new MyChatCompletionService());
+
+        // Add any other Kernel services or configurations
+        // ...
+        Kernel kernel = builder.Build();
+        // </addChatService>
+
+        return kernel;
+    }
+
+    static async Task UseTextGenerationServiceExample()
+    {
+        IKernelBuilder builder = Kernel.CreateBuilder();
+        builder.Services.AddKeyedSingleton<ITextGenerationService>("myTextService", new MyTextGenerationService());
+        Kernel kernel = builder.Build();
+
+        // <useTextService>
+        var executionSettings = new PromptExecutionSettings
+        {
+            // Add execution settings, such as the ModelID and ExtensionData
+            ModelId = "MyModelId",
+            ExtensionData = new Dictionary<string, object> { { "MaxTokens", 500 } }
+        };
+
+        // Send a prompt to your model directly through the Kernel
+        string prompt = "Please list three services offered by Azure";
+        string? response = await kernel.InvokePromptAsync<string>(prompt);
```

Comment:
OK, more an API question, but it seems odd to me that this could return `null`. Maybe explain what conditions would produce a null response?

Reply:
Unfortunately, the documentation doesn't specify when a null value is returned: [Ref Docs](https://learn.microsoft.com/en-us/dotnet/api/microsoft.semantickernel.kernelextensions.invokepromptasync?view=semantic-kernel-dotnet#microsoft-semantickernel-kernelextensions-invokepromptasync-1(microsoft-semantickernel-kernel-system-string-microsoft-semantickernel-kernelarguments-system-string-microsoft-semantickernel-iprompttemplatefactory)).

@elbruno do you know what might make the return null?

Reply:
My initial guess is that `InvokePromptAsync<TValue>(prompt)` can return null if the output of the model can't be parsed as `TValue`

Reply:
I'm testing locally this to understand more about it. I also think that if the model can't be accessed, it will return null. We can add a note here about it.

Reply:
Added a comment to explain that! Thanks!!